### PR TITLE
Fixes #17758: rtf upgrade rudder doesn't work

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/17758

### CODE REVIEW

File: vagrant.rb [hunk 1]
```diff
@@ -170,7 +170,7 @@
 $SKIP_IP ||= 1
 
 # Configure a complete platform by just providing an id and a json file
-def platform(config, pf_id, pf_name, override={})
+def platform(config, pf_id, pf_name, override={}, upgrade=True)
```
Comment (Fdall) [inline vagrant.rb:173]: I am almost sure it should be False by default, and the Platform object instanciated by the parser when calling the "update-rudder" cli should set the boolean to true

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
PR updated with a new commit

**Pull request merged**